src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -209,11 +212,20 @@ internal ItemManager(RecentItemsManager recentItemsManager)
                 return null;
             }
 
-            if (initialListOfItemsToBeIncluded.Count == 0)
+            if (builder.Count == 0)
             {
                 return HandleAllItemsFilteredOut(reason, data.SelectedFilters, completionRules);
             }
 
+            var initialListOfItemsToBeIncluded = builder.ToImmutableArray();
+
+            var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+            if (experimentationService.IsExperimentEnabled(WellKnownExperimentNames.SortCompletionListByMatch))
+            {
+                // Need a stable sorting algorithm to preserve the original order for items with same pattern match score.
+                initialListOfItemsToBeIncluded = initialListOfItemsToBeIncluded.OrderBy(new NullablePatternMatchComparer()).ToImmutableArray();
```

Comment:
>initialListOfItemsToBeIncluded = initialListOfItemsToBeIncluded.OrderBy(new NullablePatternMatchComparer()).ToImmutableArray(); [](start = 16, length = 127)

nit: maybe to sort the builder in the case to reduce two ToImmutableArray() calls

Reply:
Fixed